Export sync config

Update exports to fix the following error:

```
Caused by: java.lang.IllegalAccessException: class com.swirlds.config.impl.internal.ConfigDataFactory (in module com.swirlds.config.impl) cannot access class com.swirlds.platform.gossip.sync.config.SyncConfig (in module com.swirlds.platform) because module com.swirlds.platform does not export com.swirlds.platform.gossip.sync.config to module com.swirlds.config.impl
	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:392)
	at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:674)
```

--- a/platform-sdk/swirlds-platform-core/src/main/java/module-info.java
+++ b/platform-sdk/swirlds-platform-core/src/main/java/module-info.java
@@ -111,6 +111,8 @@ module com.swirlds.platform {
             com.swirlds.platform.test;
     exports com.swirlds.platform.uptime to
             com.swirlds.config.impl;
+    exports com.swirlds.platform.gossip.sync.config to
+            com.swirlds.config.impl;
 
     opens com.swirlds.platform.cli to
             info.picocli;

